cert/sign.go

```diff
@@ -76,15 +76,89 @@ func (t *TBSCertificate) Sign(signer Certificate, curve Curve, key []byte) (Cert
 	}
 }
 
+// readyToSign checks all signing requirements that don't require us to cross-reference with a CA
+func (t *TBSCertificate) readyToSign() error {
```

Comment:
I had started on this validation in `cert_v1.go` and `cert_v2.go` specifically that way we don't accidentally leak rules from one version to the other that shouldn't apply. We also need to do these things at unmarshal (or at a minimum at handshake and cert loading time) to ensure we are working with a valid certificate.

Reply:
ah yeah nice catch, I'll move this